### DISCUSSION TITLE
Basic microtonality

### DISCUSSION THREAD

**Discussion by sissyfus** [SynthstromAudible/DelugeFirmware#616] [Ideas]
I am thinking of the ability to finetune the steps of the scales of samples and synths, not a full on scale file support. If implemented it would make possible most folk music scales, potentially making it usable for oriental music and also nordic music with just intonation etc. 

Again I do not know about the technical difficulties of making such a feature, just  throwing out a wish as a composer and heavy Deluge user!

**Comment by sapphire-arches** [comment]
How do you imagine the UI for this working? There's also some technical challenges, but those are Just Work :tm:, and something that others have been working on relaxing for e.g. accidental support :smile:.

**Reply by sissyfus** [reply]
I would think of someting like when you sharpen or flatten a note in a scale by pressing audition and turn select. Press audition and Push-turn or shift-turn select to finetune the steps in the scale?

If this works additional microtonal scale presets would also be amazing but finetuning steps is what I am searching for personally.

**Comment by davidvacek** [comment]
@bobtwinkles 

Hello I would like to add to this topic. I do not know about technical dificulties, but...

I would love to see microtonal capability and overall update to scales and tunings as of now it is very basic and limited - creation of custom scales or implementing / importing scala files. Unlocking the possibilities for easy workflows beyond 12-TET tuning. This could also create possibility to send microtonal messages from Deluge to MIDI or CV/eurorack to control other devices with microtonality.

I see it as a big major update not only for people that want to explore odd tunings, but also for those that live in the areas that using these tunings or try to replicate / use instruments for specific tunings.

### **Implementation quick ideas:**

1. first and the biggest one would be importing SCALA files over sysex or just having them in the folder structure and loading them in

2. another is to use microtuning capability of MIDI standard to transmit / use deluge as source of microtonal controler for other instruments

3. translating microtonal to CV to have possibility to control eurorack and other CV capable gear.

4. individual scale presets for individual instruments / presets / rows on deluge

### **Basic Operation**
Whole system could "replace" actual scale button / functionality

**scale**
= turns scale on or of as it is now

**shift + scale**
= switching/rotating between preset scales + custom ones: Aeolian, locrian..., ..., custtom 1, custom2, custom3 etc... (with possibility to name the custom ones)

**scale+load**
= loads up scala tunning file from the SD card to the "custom" slot of the scale

### **customize tunnig on deluge**

**learn + scale** - switches the grid to custom tuning layout - something like layouts for Song view, Arranger view etc...
= every grid button represents the note 0-127 that you can manualy tune to your desired frequency

**turn select knob**
= defines how many notes (1-128) will be in the octave/custom scale, according number of grid buttons will be lit up for editing them  - default 12

**shift+grid button** 
= defines the root note - default A

**shift +turn select knob**
defines root/reference frequency - default 440hz (A440)
 
**hold grid button + turn select knob**
= adjusting tuning for the note by cents - default 50cents / 1semitone

**hold grid button + turn up/dn left/right knobs**
= possibility to use ratios - 5/2, 4/7 etc.... by turning up/dn knob you change first number in ratio by turning left/right knob you will change seccond number for the ratio

**scale+save**
= saves the scala format to the SD card
